src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.PullMemberUp;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Roslyn.Utilities;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+using EditorMap = System.Collections.Immutable.ImmutableDictionary<Microsoft.CodeAnalysis.SyntaxTree, Microsoft.CodeAnalysis.Editing.DocumentEditor>;
+using SyntaxMap = System.Collections.Immutable.ImmutableDictionary<Microsoft.CodeAnalysis.ISymbol, Microsoft.CodeAnalysis.SyntaxNode[]>;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal class MembersPuller
+    {
+        internal static readonly MembersPuller Instance = new MembersPuller();
+
+        private MembersPuller()
+        {
+        }
+
+        /// <summary>
+        /// Return the CodeAction to pull selectedMember up to destinationType. If the pulling will cause error,
```

Comment:
Use paramref for selectedMember so if it gets renamed it doesn't get out of sync.